sdk/core/System.Net.ClientModel/api/System.Net.ClientModel.net6.0.cs

```diff
@@ -0,0 +1,407 @@
+        public abstract System.Net.ClientModel.Core.RequestBodyContent? Content { get; set; }
+        public abstract System.Net.ClientModel.Core.MessageHeaders Headers { get; }
+        public abstract string Method { get; set; }
+        public abstract System.Uri Uri { get; set; }
+        public abstract void Dispose();
+    }
+    public abstract partial class MessageResponse : System.IDisposable
+    {
+        protected MessageResponse() { }
+        public System.BinaryData Content { get { throw null; } }
+        public abstract System.IO.Stream? ContentStream { get; protected internal set; }
+        public abstract System.Net.ClientModel.Core.MessageHeaders Headers { get; }
+        public bool IsError { get { throw null; } }
+        public abstract string ReasonPhrase { get; }
+        public abstract int Status { get; }
+        public abstract void Dispose();
+    }
+    [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute("The constructors of the type being deserialized are dynamically accessed and may be trimmed.")]
+    public partial class ModelJsonConverter : System.Text.Json.Serialization.JsonConverter<System.Net.ClientModel.Core.IJsonModel<object>>
+    {
+        public ModelJsonConverter() { }
+        public ModelJsonConverter(System.Net.ClientModel.ModelReaderWriterFormat format) { }
+        public ModelJsonConverter(System.Net.ClientModel.ModelReaderWriterOptions options) { }
+        public System.Net.ClientModel.ModelReaderWriterOptions ModelReaderWriterOptions { get { throw null; } }
+        public override bool CanConvert(System.Type typeToConvert) { throw null; }
+        public override System.Net.ClientModel.Core.IJsonModel<object> Read(ref System.Text.Json.Utf8JsonReader reader, System.Type typeToConvert, System.Text.Json.JsonSerializerOptions options) { throw null; }
+        public override void Write(System.Text.Json.Utf8JsonWriter writer, System.Net.ClientModel.Core.IJsonModel<object> value, System.Text.Json.JsonSerializerOptions options) { }
+    }
+    [System.AttributeUsageAttribute(System.AttributeTargets.Class)]
+    public sealed partial class ModelReaderProxyAttribute : System.Attribute
+    {
+        public ModelReaderProxyAttribute([System.Diagnostics.CodeAnalysis.DynamicallyAccessedMembersAttribute(System.Diagnostics.CodeAnalysis.DynamicallyAccessedMemberTypes.NonPublicConstructors | System.Diagnostics.CodeAnalysis.DynamicallyAccessedMemberTypes.PublicConstructors)] System.Type proxyType) { }
+        [System.Diagnostics.CodeAnalysis.DynamicallyAccessedMembersAttribute(System.Diagnostics.CodeAnalysis.DynamicallyAccessedMemberTypes.NonPublicConstructors | System.Diagnostics.CodeAnalysis.DynamicallyAccessedMemberTypes.PublicConstructors)]
+        public System.Type ProxyType { get { throw null; } }
+    }
+    public abstract partial class PipelineEnumerator
+    {
+        protected PipelineEnumerator() { }
+        public abstract int Length { get; }
```

Comment:
Should we call this an Enumerator if it has a Length property and doesn't implement `IEnumerable`?

Reply:
Yeah, it is a bit strange. Should we call it PipelineProcessor?

Reply:
It did get questions from the BCL team - I like this suggestion and will rename